jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/PathMappingsHandler.java

```diff
@@ -0,0 +1,70 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import org.eclipse.jetty.http.pathmap.MatchedResource;
+import org.eclipse.jetty.http.pathmap.PathMappings;
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Handler that delegates to other handlers through a configured {@link PathMappings}.
+ */
+public class PathMappingsHandler extends Handler.Wrapper
```

Comment:
I don't think this is a `Handler.Wrapper`, rather it is a `Handler.Container`

Reply:
Hmmm I guess it can be both?

Reply:
Eitherway, it needs to respect the `getHandlers()` and `getDescendants()` contracts.  These might be simpler to consider if this was just a `Container` and not a `Wrapper`.  You can always use the "/" or "/*" mappings to get a default mapping.

Reply:
Yeah, I think this should extend `Handler.Collection` and be named `PathMappedHandlerCollection`

Reply:
@gregw `Handler.Collection` has a TODO to be renamed `Handler.List`. 

I think this new class should extend `Handler.AbstractContainer` because it's not a list nor a collection (it's more like a map) and just be called `PathMappingsHandler` or perhaps `PathMappingsHandlerContainer`, but nothing related to `List` or `Collection`.